activesupport/test/dependencies_test.rb

```diff
@@ -185,6 +185,44 @@ def test_require_dependency_interaction_with_autoloading
     end
   end
 
+  # Regression see https://github.com/rails/rails/issues/31694
+  def test_included_constant_that_changes_to_have_exception_then_back_does_not_loop_forever
```

Comment:
I reverted the change locally and this test is passing.

Reply:
🤔